### PR TITLE
Support for directories

### PR DESCRIPTION
**Pull request by juanolon**
With this change, sub-directories are supported. For my usecase, i have folders for specific topics, which are unrelated to my other notes (e.g. uni courses). I have been using it already for a week, and it works well. Whenever a folder is open, the site is redirected to the `index` page. If there is no one, then as usual, the user is redirected to the edit page. I find it okey so, but one could think of having a folder template? what do you think?

### CODE REVIEW

**Comment by emad-elsaid** [comment]
I'm not following the problem that this PR tries to solve. can you explain a little the usecase, how it works now and how you would like it to work, then what is the solution you implemented here?

File: handlers.go [hunk 1]
```diff
@@ -59,6 +59,10 @@ func getPageHandler(w Response, r Request) Output {
 	page := NewPage(r.PathValue("page"))
 
 	if !page.Exists() {
+		if s, err := os.Stat(page.Name()); err == nil && s.IsDir() {
+			// TODO now it redirects to index (md or html). check if the files exists, otherwise render default template. template should just print all childrent notes
+			return Redirect("index")
```
Comment (emad-elsaid) [inline handlers.go:64]: I think this shouldn't be "index" but the constant `INDEX` as the user can change the index file name from `index` to `README` to `home`

**Comment by juanolon** [comment]
To keep my notes separated, I have different folders for my courses. I could think of having different gardens, because they are totally unrelated.. anyway,  let say i have the following folder structure:
```
- note-a
- note-b
folder-a/
    - note-a-a
    - note-a-b
folder-b
    - note-b-a
    - note-b-b
```
xlog did render `/note-a` and `folder-a/note-a-a`, but it was cumbersome to browse my notes, because i can't just open `folder-a/` on the browser and have an overview of the notes for topic `folder-a`. (Actually, xlog returns a "static" page showing **all** files under that folder. When visiting any of the listed files, the file is downloaded - even if it is a markdown file)
Also, hot-loading was only working for notes on the root directory (fsnotify/fsnotify doesn't support for recursive file watching).
With this changes, 1) hot-loading works on file changes in sub directories 2) whenever a folder is visited, you get redirected to the `INDEX` page.

File: handlers.go (hunk 1 above)
Comment (juanolon) [reply handlers.go:64]: that makes totally sense!

**Review by emad-elsaid — APPROVED** [review]

**Comment by emad-elsaid** [comment]
The part that serve the directory actually can be some extension that renders a directory content similar to blogs or just titles...etc 
I imagine an extension that provides Pages Source, which will return with a page when it's a directory. should be straightforward to do it

**Pull request merged**